websites/E/e621/presence.ts

```diff
@@ -171,40 +157,41 @@ presence.on("UpdateData", async () => {
 				];
 			}
 			search = document.querySelector("#search_name_matches");
-			if (search.value !== "") {
-				presenceData.details = "Searching Pools For:";
+			if (!search.value) {
+				presenceData.details = "Searching Pools for:";
+				presenceData.smallImageKey = "search";
 				presenceData.state = search.value;
 			} else presenceData.details = "Pools";
 		} else if (path.includes("post_sets")) presenceData.details = "Post Sets";
 		else if (path.includes("wiki_pages")) {
-			if (buttons) {
-				presenceData.buttons = [
-					{
-						label: "View Wiki",
-						url: document.location.href
-					}
-				];
-			}
+			presenceData.buttons = [
+				{
+					label: "View Wiki",
+					url: document.location.href
+				}
+			];
 			search = document.querySelector("#quick_search_title");
-			if (search.value !== "") {
-				presenceData.details = "Searching Wiki For:";
+			if (search.value) {
+				presenceData.details = "Searching Wiki for:";
+				presenceData.smallImageKey = "search";
 				presenceData.state = search.value;
 			} else {
 				title = document.querySelector("#wiki-page-title > a");
-				presenceData.details = title.textContent;
+				presenceData.details = "Reading Wiki:";
+				presenceData.state = title.textContent;
+				presenceData.smallImageKey = "reading";
 			}
 		} else if (path.includes("forum_topics/")) {
-			if (buttons) {
-				presenceData.buttons = [
-					{
-						label: "View Forum Post",
-						url: document.location.href
-					}
-				];
-			}
+			presenceData.buttons = [
+				{
+					label: "View Forum Post",
+					url: document.location.href
+				}
+			];
 			search = document.querySelector("#quick_search_body_matches");
-			if (search.value !== "") {
-				presenceData.details = "Searching Forum For:";
+			if (!search.value) {
```

Comment:
That's not what the old if was checking
```suggestion
			if (search.value) {
```